src/inference/graph_inference.cpp

```diff
@@ -351,6 +351,54 @@ class GraphNodeAssignment : public Nice2Assignment {
     }
   }
 
+  std::vector<std::pair<int, double>> GetCandidatesForNode(Nice2Inference& inference, int node) {
+        GraphInference& graphInference = static_cast<GraphInference&>(inference);
+        std::vector<int> candidates;
+        candidates.clear();
+        GetLabelCandidates(graphInference, node, &candidates, kMaxPerArcBeamSize);
+
+        std::vector<std::pair<int, double>> scoredCandidates;
```

Comment:
If you want to save reallocating this, you can pass a pointer of it as a parameter instead of making the return value of the function. Then, GetCandidatesForNode() clears it and updates the candidates.